File: Medium Problems/maximumTripletValue.py
class Solution:
    def maximumTripletValue(self, nums: List[int]) -> int:
        next_biggest = list(reversed(list(accumulate(reversed(nums),max))))
        biggest_seen = nums[0]
        ans = 0
        for n in range(1,len(nums)-1):
            ans = max(ans, (biggest_seen - nums[n]) * next_biggest[n+1])
            biggest_seen = max(biggest_seen, nums[n])
        return ans

File: Medium Problems/maximum_triplet_value.rs
impl Solution {
    pub fn maximum_triplet_value(nums: Vec<i32>) -> i64 {
        let mut dif = 0 as i64;
        let mut max_seen= nums[0] as i64;
        let mut ans = 0 as i64;
        let mut num;
        for n in nums{
            num = n as i64;
            ans = ans.max(dif * num);
            dif = dif.max(max_seen - num);
            max_seen = max_seen.max(num as i64);
        }
        ans
        
        
    }
}
